pkg/controller/controlplane/valuesprovider.go

```diff
@@ -370,8 +379,17 @@ func (vp *valuesProvider) GetControlPlaneChartValues(
 		return nil, err
 	}
 
+	// TODO: this is a workaround to speed things for the time being...
+	// the infrastructure controller writes the nodes cidr back into the infrastructure status, but the cluster resource does not contain it immediately
+	// it would need the start of another reconcilation until the node cidr can be picked up from the cluster resource
+	// therefore, we read it directly from the infrastructure status
+	infrastructure, err := vp.gardenerExtensionsClientset.ExtensionsV1alpha1().Infrastructures(cluster.Seed.Namespace).Get(cluster.Shoot.Name, metav1.GetOptions{})
```

Comment:
Why don't you use the client that you already have?

```suggestion
  infrastructure := &extensionsv1alpha1.Infrastructure{}
  if err := vp.client.Get(ctx, kutil.Key(cp.Namespace, cp.Name); infrastructure); err != nil {
    return nil, err
  }
```

Reply:
Oh... I thought with this one you could only get Kubernetes objects. That looks of course much better. Thank you.